src/js/claims-status/containers/YourClaimsPage.jsx

```diff
@@ -118,27 +118,49 @@ class YourClaimsPage extends React.Component {
   }
 
   render() {
-    const { unfilteredAppeals, unfilteredClaims, list, pages, page, loading, show30DayNotice, route, synced } = this.props;
+    const {
+      unfilteredAppeals,
+      unfilteredClaims,
+      list,
+      pages,
+      page,
+      claimsLoading,
+      appealsLoading,
+      show30DayNotice,
+      route,
+      synced
+    } = this.props;
+
     const tabs = [
       'OpenClaims',
       'ClosedClaims'
     ];
 
     let content;
     let innerContent;
+    const bothRequestsLoaded = !claimsLoading && !appealsLoading;
+    const bothRequestsLoading = claimsLoading && appealsLoading;
+    const atLeastOneRequestLoading = claimsLoading || appealsLoading;
+    const emptyList = !list || !list.length;
 
-    if (loading) {
-      content = <LoadingIndicator message="Loading a list of your claims and appeals..." setFocus/>;
+    if (bothRequestsLoading || (atLeastOneRequestLoading && emptyList)) {
+      content = <LoadingIndicator message="Loading your claims and appeals..." setFocus/>;
     } else {
-      if (list.length > 0) {
+      if (!emptyList) {
         innerContent = (<div>
           {!route.showClosedClaims && show30DayNotice && <ClosedClaimMessage claims={unfilteredClaims.concat(unfilteredAppeals)} onClose={this.props.hide30DayNotice}/>}
           <div className="claim-list">
+            {atLeastOneRequestLoading &&
+              <div>
+                <LoadingIndicator message="Loading your claims and appeals..."/>
+                <br/>
+              </div>
+            }
             {list.map(claim => this.renderListItem(claim))}
             <Pagination page={page} pages={pages} onPageSelect={this.changePage}/>
           </div>
         </div>);
-      } else if (!this.props.canAccessClaims) {
+      } else if (!this.props.canAccessClaims && bothRequestsLoaded) {
```

Comment:
Is it possible to have appeals if `!this.props.canAccessClaims`?

Reply:
I believe it's definitely possible, but in this conditional the first statement on top should catch it (having appeals would mean that the list is not empty).